backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -417,3 +418,84 @@ def handle_revoke_failure(
             table, self.share, self.target_environment
         )
         return True
+
+    def handle_share_failure_for_all_tables(self, tables, error, share_item_status):
+        for table in tables:
+            share_item = ShareObjectRepository.find_sharable_item(
+                self.session, self.share.shareUri, table.tableUri
+            )
+            share_item_sm = ShareItemSM(share_item_status)
+            new_state = share_item_sm.run_transition(ShareObjectActions.Start.value)
+            share_item_sm.update_state_single_item(self.session, share_item, new_state)
+            new_state = share_item_sm.run_transition(ShareItemActions.Failure.value)
+            share_item_sm.update_state_single_item(self.session, share_item, new_state)
+
+            self.handle_revoke_failure(table=table, error=error)
```

Comment:
The function `handle_share_failure_for_all_tables` is used both in the approve and in the revoke processes but it calls handle_revoke_failure for both processes

Reply:
Will update this . Thanks @dlpzx